compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -852,13 +852,17 @@ object SymDenotations {
      * During completion, references to moduleClass and sourceModules are stored in
      * the completers.
      */
-    /** The class implementing this module, NoSymbol if not applicable. */
+    /** If this a module class, return itself, if this is a module, the corresponding class,
+     *  otherwise NoSymbol
+     */
     final def moduleClass(implicit ctx: Context): Symbol = {
       def notFound = {
       	if (Config.showCompletions) println(s"missing module class for $name: $myInfo")
       	NoSymbol
       }
-      if (this is ModuleVal)
+      if (this is ModuleClass)
```

Comment:
For speed I'd move the operations that return the same symbol towards the end, i.e. where otherwise `NoSymbol` would be returned.